### PR TITLE
Fix Quartz rescheduling with multiple triggers

### LINKED ISSUE #50063: 0.51.x: ERROR metabase.task Error rescheduling job,org.quartz.ObjectAlreadyExistsException
### Describe the bug

We upgraded from 0.50.x to 0.51.x and these started occurring. It also seems that new alerts are not being sent, unclear if related.

### To Reproduce

1. Upgrade to 0.51.x

### Expected behavior

No error is reported.

### Logs

```
[fa705b8e-d7fb-4d33-b15b-74494d6383e9] 2024-11-15T13:17:37+11:00 ERROR metabase.task Error rescheduling job
org.quartz.ObjectAlreadyExistsException: Unable to store Trigger with name: 'metabase.task.backfill-query-fields.trigger' and group: 'DEFAULT', because one already exists with this identification.
at org.quartz.impl.jdbcjobstore.JobStoreSupport.storeTrigger(JobStoreSupport.java:1184)
at org.quartz.impl.jdbcjobstore.JobStoreSupport.replaceTrigger(JobStoreSupport.java:1503)
at org.quartz.impl.jdbcjobstore.JobStoreSupport$11.execute(JobStoreSupport.java:1478)
at org.quartz.impl.jdbcjobstore.JobStoreSupport.executeInNonManagedTXLock(JobStoreSupport.java:3864)
at org.quartz.impl.jdbcjobstore.JobStoreTX.executeInLock(JobStoreTX.java:93)
at org.quartz.impl.jdbcjobstore.JobStoreSupport.replaceTrigger(JobStoreSupport.java:1474)
at org.quartz.core.QuartzScheduler.rescheduleJob(QuartzScheduler.java:1121)
at org.quartz.impl.StdScheduler.rescheduleJob(StdScheduler.java:321)
at metabase.task$reschedule_task_BANG_93209__93210.invokeStatic(task.clj:201)
at metabase.task$reschedule_task_BANG_93209__93210.invoke(task.clj:192)
at metabase.task$schedule_task_BANG_93225__93226.invokeStatic(task.clj:221)
at metabase.task$schedule_task_BANG_93225__93226.invoke(task.clj:213)
at metabase.task.sweep_query_analysis$fn__108080.invokeStatic(sweep_query_analysis.clj:113)
at metabase.task.sweep_query_analysis$fn__108080.invoke(sweep_query_analysis.clj:96)
at metabase.task$init_tasks_BANG_$fn__93132.invoke(task.clj:79)
at metabase.task$init_tasks_BANG_.invokeStatic(task.clj:75)
at metabase.task$init_tasks_BANG_.invoke(task.clj:71)
at metabase.task$init_scheduler_BANG_.invokeStatic(task.clj:166)
at metabase.task$init_scheduler_BANG_.invoke(task.clj:153)
at metabase.task$start_scheduler_BANG_.invokeStatic(task.clj:177)
at metabase.task$start_scheduler_BANG_.invoke(task.clj:172)
at metabase.core$init_BANG__STAR_.invokeStatic(core.clj:166)
at metabase.core$init_BANG__STAR_.invoke(core.clj:102)
at metabase.core$init_BANG_.invokeStatic(core.clj:181)
at metabase.core$init_BANG_.invoke(core.clj:176)
at metabase.core$start_normally.invokeStatic(core.clj:193)
at metabase.core$start_normally.invoke(core.clj:187)
at metabase.core$entrypoint.invokeStatic(core.clj:226)
at metabase.core$entrypoint.doInvoke(core.clj:220)
at clojure.lang.RestFn.invoke(RestFn.java:400)
at clojure.lang.AFn.applyToHelper(AFn.java:152)
at clojure.lang.RestFn.applyTo(RestFn.java:135)
at clojure.lang.Var.applyTo(Var.java:707)
at clojure.core$apply.invokeStatic(core.clj:667)
at clojure.core$apply.invoke(core.clj:662)
at metabase.bootstrap$_main.invokeStatic(bootstrap.clj:31)
at metabase.bootstrap$_main.doInvoke(bootstrap.clj:28)
at clojure.lang.RestFn.invoke(RestFn.java:400)
at clojure.lang.AFn.applyToHelper(AFn.java:152)
at clojure.lang.RestFn.applyTo(RestFn.java:135)
at metabase.bootstrap.main(Unknown Source)
```

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-AU",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:132.0) Gecko/20100101 Firefox/132.0",
    "vendor": ""
  },
  "metabase-info": {
    "databases": [
      "mysql",
      "postgres",
      "redshift"
    ],
    "run-mode": "prod",
    "plan-alias": "",
    "version": {
      "date": "2024-11-11",
      "tag": "v0.51.3",
      "hash": "d757d0b"
    },
    "settings": {
      "report-timezone": "Australia/Sydney"
    },
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "14.10"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.3"
      }
    }
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.25+9",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.25",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.25+9",
    "os.name": "Linux",
    "os.version": "5.10.226-214.880.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "Australia/Sydney"
  }
}
```

### Severity

blocking some users

### Additional context

Deployment context is Docker using AWS ECS Fargate, 1-2 instances typically running concurrently.

### PR DESCRIPTION
**Pull request by crisptrutski**
Fixes https://github.com/metabase/metabase/issues/50063

Stats is having a sad where the first trigger we pull out from Quartz is *not* the one with the conflicting trigger, and then we get another conflict exception when trying to swap them out.

This updated behavior is strictly better than before, but it's still not perfect. Ideally we'd have one method that takes *all* the triggers in one go, atomically swaps out all the matching entries, and deletes the other obsolete records. I don't think we make enough use of multi-trigger jobs right now that this is important.

This also fixes how we handle misfires for the query analysis sweeper - I think it's safer to skip a firing than to run non-stop.

I also snuck in a little change to make Sanya happy about honesty in log messages.

See: [registration failure logs](https://grafana.monitoring.metabase.com/d/Zjw_nJX7k/metabase-logs?orgId=1&refresh=5m&var-cluster=staging-1&var-filter=Unable%20to%20store%20Trigger%20with%20name&var-namespace=All&var-dns_host=stats.metabase.com&var-pod=All&from=now-1h&to=now)

### CODE REVIEW

File: src/metabase/task/sweep_query_analysis.clj [hunk 1]
```diff
@@ -98,18 +98,14 @@
         job     (jobs/build
                  (jobs/of-type SweepQueryAnalysis)
                  (jobs/with-identity job-key)
-                 (jobs/store-durably)
-                 (jobs/request-recovery))
```
Comment (crisptrutski) [inline src/metabase/task/sweep_query_analysis.clj:102]: The job is idempotent, just let the next firing take care of things.

File: src/metabase/task/sweep_query_analysis.clj [hunk 2]
```diff
@@ -98,18 +98,14 @@
...
+                 (jobs/store-durably))
         trigger (triggers/build
                  (triggers/with-identity (triggers/key "metabase.task.backfill-query-fields.trigger"))
-                 (triggers/start-now)
```
Comment (crisptrutski) [inline src/metabase/task/sweep_query_analysis.clj:105]: This kind of compound trigger does not seem to work properly, stick to `trigger-now!`, which creates another temporary trigger.

**Review by qnkhuat — APPROVED** [review]
lgtm! it would be great if you add a test for `reschedule-task!` while you're at it :)

**Comment by crisptrutski** [comment]
> lgtm! it would be great if you add a test for `reschedule-task!` while you're at it :)

I'll save testing it for when I improve it further 😄

**Comment by crisptrutski** [comment]
@metabase-bot backport release-x.51.x

**Pull request merged**